src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -500,6 +597,144 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+                    {
+                        // We can have duplicates because arguments can be duplicated, but we
+                        // only need to register one placeholder per argument
+                        if (HasPlaceholderReplacement(placeholder))
+                        {
+                            continue;
+                        }
+
+                        switch (placeholder.ArgumentIndex)
+                        {
+                            case BoundInterpolatedStringArgumentPlaceholder.InstanceParameter:
+                                Debug.Assert(requiresInstanceReceiver);
+                                Debug.Assert(visitedReceiver != null);
+                                AddPlaceholderReplacement(placeholder, visitedReceiver);
+                                break;
+                            case BoundInterpolatedStringArgumentPlaceholder.TrailingConstructorValidityParameter:
+                                Debug.Assert(trailingOutVariable != null);
+                                AddPlaceholderReplacement(placeholder, trailingOutVariable);
+                                break;
+                            case >= 0 and var argumentIndex:
+                                Debug.Assert(argumentIndex < interpolatedStringHandlerArgumentIndex);
+                                var placeholderParamIndex = argsToParamsOpt.IsDefault ? argumentIndex : argsToParamsOpt[argumentIndex];
+                                Debug.Assert(actualArguments[placeholderParamIndex] is not null);
+                                AddPlaceholderReplacement(placeholder, actualArguments[placeholderParamIndex]);
+                                break;
+                            default:
+                                throw ExceptionUtilities.UnexpectedValue(placeholder.ArgumentIndex);
+                        }
+                    }
+
+                    var (handlerExpressions, result) = RewriteToInterpolatedStringHandlerPattern(interpolatedString);
+
+                    foreach (var placeholder in interpolationData.ArgumentPlaceholders)
+                    {
+                        RemovePlaceholderReplacementIfPresent(placeholder);
+                    }
+
+                    // We create a sequence, rather than just directly adding all the expressions to storesToTemps, because this
+                    // sequence of handler expressions could itself contain temp stores and won't be able to differentiate them
+                    // from the temporaries owned by this lowering when we're later creating the list of temporaries for this
+                    // call.
+                    var handlerSequence = _factory.Sequence(
+                        ImmutableArray<LocalSymbol>.Empty,
+                        handlerExpressions.ToImmutableAndFree(),
+                        result);
+                    storesToTemps.Add(handlerSequence);
```

Comment:
>handlerSequence [](http://example.com/codeflow?start=38&length=15)

I am not sure why do we need to add this sequence to storesToTemps, especially with ```result``` in it.